File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ExpectedExceptionAttribute.vb
Imports System

Public Class ExceptionTests

    <ExpectedException()> 'Noncompliant {{Use an Assert method to test the thrown exception.}}
    Public Sub ExpectedExceptionAttrbutesShouldNotBeUsed()
    End Sub

    <ExpectedException(GetType(DivideByZeroException))> ' Noncompliant
    Public Sub ExpectedExceptionAttrbutesShouldNotBeUsedWithType()
    End Sub
End Class

Public Class ExpectedExceptionAttribute
    Inherits Attribute
    Public Sub New()
    End Sub
    Public Sub New(expectedType As Type)
        ExceptionType = expectedType
    End Sub
    Public Property ExceptionType As Type
End Class

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ExpectedExceptionAttribute.cs
using System;

public class ExceptionTests
{
    [ExpectedException] // Noncompliant {{Use an Assert method to test the thrown exception.}}
//   ^^^^^^^^^^^^^^^^^
    public void ExpectedExceptionAttrbutesShouldNotBeUsed() { }

    [ExpectedException(typeof(DivideByZeroException))] // Noncompliant
    public void ExpectedExceptionAttrbutesShouldNotBeUsedWithType() { }
}

public class ExpectedExceptionAttribute : Attribute
{
    public ExpectedExceptionAttribute(Type expectedType = null)
    {
        ExceptionType = expectedType;
    }
    public Type ExceptionType { get; }
}
